esprima.js

```diff
@@ -4023,18 +4023,20 @@
         if (match('=')) {
             lex();
             init = isolateCoverGrammar(parseAssignmentExpression);
-        } else if (id.type !== Syntax.Identifier) {
-            expect('=');
+        } else {
+            if (!(options && options.inFor) && (id.type !== Syntax.Identifier)) {
```

Comment:
This is for pure readability. It's challenging to go through that logic without extra help.